dataland-frontend/src/components/forms/CreateSFDRDataset.vue

```diff
@@ -0,0 +1,309 @@
+    const dataId = this.route.query.templateDataId;
+    if (dataId && typeof dataId === "string") {
+      void this.loadSfdrData(dataId);
+    } else {
+      this.waitingForData = false;
+    }
+  },
+  methods: {
+    /**
+     * Loads the SFDR-Dataset identified by the provided dataId and pre-configures the form to contain the data
+     * from the dataset
+     * @param dataId the id of the dataset to load
+     */
+    async loadSfdrData(dataId: string): Promise<void> {
+      this.waitingForData = true;
+      const sfdrDataControllerApi = await new ApiClientProvider(
+        assertDefined(this.getKeycloakPromise)(),
+      ).getSfdrDataControllerApi();
+
+      const dataResponse = await sfdrDataControllerApi.getCompanyAssociatedSfdrData(dataId);
+      const sfdrResponseData = dataResponse.data;
+      this.referencedReportsForPrefill = sfdrResponseData.data.general.general.referencedReports;
+      this.companyAssociatedSfdrData = objectDropNull(sfdrResponseData) as CompanyAssociatedDataSfdrData;
+
+      this.waitingForData = false;
+    },
+    /**
+     * Sends data to add SFDR data
+     */
+    async postSfdrData(): Promise<void> {
+      this.messageCounter++;
+      try {
+        if (this.documents.size > 0) {
+          await uploadFiles(Array.from(this.documents.values()), assertDefined(this.getKeycloakPromise));
+        }
+
+        const sfdrDataControllerApi = await new ApiClientProvider(
+          assertDefined(this.getKeycloakPromise)(),
+        ).getSfdrDataControllerApi();
+        await sfdrDataControllerApi.postCompanyAssociatedSfdrData(this.companyAssociatedSfdrData);
+        this.$emit("datasetCreated");
+        this.dataDate = undefined;
+        this.message = "Upload successfully executed.";
+        this.uploadSucceded = true;
+      } catch (error) {
+        console.error(error);
+        if (error instanceof AxiosError) {
+          this.message = "An error occurred: " + error.message;
+        } else {
+          this.message =
+            "An unexpected error occurred. Please try again or contact the support team if the issue persists.";
+        }
+        this.uploadSucceded = false;
+      } finally {
+        this.postSfdrDataProcessed = true;
+      }
+    },
+    //TODO refactor the name of this file to CreateSfdrDataset
+    /**
+     * updates the list of documents that were uploaded
+     * @param reportsNames repots names
+     * @param reportsToUpload reports to upload
+     */
+    updateDocumentsList(reportsNames: string[], reportsToUpload: DocumentToUpload[]) {
+      this.namesOfAllCompanyReportsForTheDataset = reportsNames;
+      this.documents = new Map();
+      reportsToUpload.forEach((document) => this.documents.set(document.file.name, document));
+    },
+  },
+  provide() {
+    return {
+      namesOfAllCompanyReportsForTheDataset: computed(() => {
+        return this.namesOfAllCompanyReportsForTheDataset;
+      }),
+      referencedReportsForPrefill: computed(() => {
```

Comment:
Why has this been done via "provide", instead of passing it down via props?

Reply:
Ok makes totally sense. It should only be accessible to the `UploadReports` component. Will close this now.